pkg/pub_package_reader/test/package_archive_test.dart

```diff
@@ -483,4 +483,59 @@ $dependencies
       expect(requireNonEmptyLicense('LICENSE', 'BSD license'), isEmpty);
     });
   });
+
+  group('pubspec.yaml to json conversion', () {
+    test('pubspec.yaml to json conversion valid', () {
+      final pubspec = '''
```

Comment:
Let's make these examples smaller. No need for version, authors, too many dependencies...